src/util/gdal_source_datasets.cpp

```diff
@@ -12,41 +12,38 @@ const std::string suffix(".json");
 const size_t suffix_length = suffix.length();
 
 std::vector<std::string> GDALSourceDataSets::getDataSetNames() {
-    const std::string path = Configuration::get<std::string>("gdalsource.datasets.path");
+    namespace bf = boost::filesystem;
+    const bf::path path(Configuration::get<std::string>("gdalsource.datasets.path"));
 
-    struct dirent *entry;
-    DIR *dir = opendir(path.c_str());
-
-    if (dir == nullptr) {
+    if(!bf::is_directory(path))
         throw MustNotHappenException("GDAL_Service: Directory for gdal dataset files could not be found.");
-    }
 
     std::vector<std::string> dataSetNames;
 
-    while ((entry = readdir(dir)) != nullptr) {
-        std::string filename = entry->d_name;
+    for(auto it = bf::directory_iterator(path); it != bf::directory_iterator{}; ++it){
+        auto file = (*it).path();
+        if(!bf::is_regular_file(file))
+            continue;
+        auto filename = file.filename().string();
 
-        //check if file ends with the suffix
+        ///check if file ends with the suffix
         bool found = filename.find(suffix, filename.length() - suffix_length)  != std::string::npos;
```

Comment:
there should be a function on boost path to get the extension without string operation

Reply:
I changed it at the three places that compare suffixes of files.
(And a small merge conflict appeared because of that.)